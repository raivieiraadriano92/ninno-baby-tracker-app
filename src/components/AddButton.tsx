import Ionicons from "@expo/vector-icons/Ionicons";
import colors from "tailwindcss/colors";

import { Button } from "./Button";

import { useCustomThemeContext } from "src/context/CustomThemeProvider";

export const AddButton: typeof Button = ({ className, ...props }) => {
  const { theme } = useCustomThemeContext();

  return (
    <Button
      className={`h-12 p-0 w-12 ${className}`}
      customColors={[
        colors[theme.colorTokens.primary][500],
        colors[theme.colorTokens.primary][400]
      ]}
      enableShadow
      shadowColor={colors[theme.colorTokens.primary][500]}
      {...props}
    >
      <Ionicons name="add" size={24} color={colors.white} />
    </Button>
  );
};
