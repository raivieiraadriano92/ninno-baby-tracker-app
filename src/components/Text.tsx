import { FunctionComponent } from "react";

import { Text as RNText, TextProps as RNTextProps } from "react-native";

type TextProps = RNTextProps;

export const Text: FunctionComponent<TextProps> = ({
  children,
  className,
  ...props
}) => (
  <RNText
    className={`font-normal text-base text-black ${className}`}
    {...props}
  >
    {children}
  </RNText>
);
