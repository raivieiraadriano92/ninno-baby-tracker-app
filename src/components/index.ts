export { BabyProfileCard } from './baby-profile-card'

export { BabyProfileIcon } from './baby-profile-icon'

export { BabyProfilePickerBottomSheet } from './baby-profile-picker-bottom-sheet'

export type { BabyProfilePickerBottomSheetElement } from './baby-profile-picker-bottom-sheet'

export { BaseCard } from './base-card'

export { HappyBirthdayModal } from './happy-birthday-modal'

export type { HappyBirthdayModalElement } from './happy-birthday-modal'

export { Button } from './button'

export { DeleteBabyProfileBottomSheet } from './delete-baby-profile-bottom-sheet'

export type { DeleteBabyProfileBottomSheetElement } from './delete-baby-profile-bottom-sheet'

export { DeleteRecordBottomSheet } from './delete-record-bottom-sheet'

export type { DeleteRecordBottomSheetElement } from './delete-record-bottom-sheet'

export { FirstBabyProfile } from './first-baby-profile'

export { GlobalErrorBottomSheet } from './global-error-bottom-sheet'

export type { GlobalErrorBottomSheetElement } from './global-error-bottom-sheet'

export { MeasuresPickerBottomSheet } from './measures-picker-bottom-sheet'

export type { MeasuresPickerBottomSheetElement } from './measures-picker-bottom-sheet'

export { MeasuresPicker } from './measures-picker'

export { PageLoader } from './page-loader'

export { RecordCard } from './record-card'

export { RecordFiltersBottomSheet } from './record-filters-bottom-sheet-copy'

export type { RecordFiltersBottomSheetElement } from './record-filters-bottom-sheet-copy'

export { RecordIcon } from './record-icon'

export { RecordTypePickerBottomSheet } from './record-type-picker-bottom-sheet'

export type { RecordTypePickerBottomSheetElement } from './record-type-picker-bottom-sheet'

export { RecordTypePill } from './record-type-pill'

export { Text } from './text'
