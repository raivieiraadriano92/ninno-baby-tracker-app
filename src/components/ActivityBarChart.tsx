import { FunctionComponent } from "react";

import { View, ViewProps } from "react-native";
import { BarChart } from "react-native-gifted-charts";
import colors from "tailwindcss/colors";

import { Text } from "src/components/Text";
import { ActivityType } from "src/services/database/models/ActivityModel";
import { activityTypeAttributes } from "src/utils/global";

type ActivityBarChartProps = ViewProps & {
  activityType: ActivityType;
};

const barData = [
  { value: 230, label: "Jan" },
  { value: 180, label: "Feb" },
  { value: 200, label: "Mar" },
  { value: 250, label: "Apr" },
  { value: 320, label: "May" }
];

export const ActivityBarChart: FunctionComponent<ActivityBarChartProps> = ({
  activityType,
  className,
  style
}) => {
  const diaperAttributes = activityTypeAttributes[activityType];

  return (
    <View
      className={`p-4 rounded-2xl space-y-4 ${className}`}
      style={[
        {
          backgroundColor: colors[diaperAttributes.color][50]
        },
        style
      ]}
    >
      <View className="flex-row items-center justify-between">
        <Text
          className="font-medium"
          style={{ color: colors[diaperAttributes.color][500] }}
        >{`${diaperAttributes.emoji} ${diaperAttributes.title}`}</Text>
        <View className="flex-row items-center space-x-2">
          <Text
            className="text-xs"
            style={{ color: colors[diaperAttributes.color][500] }}
          >
            {"Average: "}
            <Text
              className="font-medium text-sm"
              style={{ color: colors[diaperAttributes.color][500] }}
            >
              1
            </Text>
          </Text>
          <Text
            className="text-xs"
            style={{ color: colors[diaperAttributes.color][500] }}
          >
            {"Min: "}
            <Text
              className="font-medium text-sm"
              style={{ color: colors[diaperAttributes.color][500] }}
            >
              1
            </Text>
          </Text>
          <Text
            className="text-xs"
            style={{ color: colors[diaperAttributes.color][500] }}
          >
            {"Max: "}
            <Text
              className="font-medium text-sm"
              style={{ color: colors[diaperAttributes.color][500] }}
            >
              1
            </Text>
          </Text>
        </View>
      </View>
      <View>
        <BarChart
          noOfSections={4}
          data={barData}
          isAnimated
          frontColor={colors[diaperAttributes.color][500]}
          xAxisColor={colors[diaperAttributes.color][500]}
          yAxisColor={colors[diaperAttributes.color][500]}
          xAxisLabelTextStyle={{
            color: colors[diaperAttributes.color][500]
          }}
          yAxisTextStyle={{
            color: colors[diaperAttributes.color][500]
          }}
          rulesColor={colors[diaperAttributes.color][300]}
        />
      </View>
    </View>
  );
};
