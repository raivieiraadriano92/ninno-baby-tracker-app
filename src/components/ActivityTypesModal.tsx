import { forwardRef, useImperativeHandle, useRef } from "react";

import Ionicons from "@expo/vector-icons/Ionicons";
import { TouchableOpacity, View } from "react-native";
import colors from "tailwindcss/colors";

import { Card } from "./Card";
import { Modal, ModalRef } from "./Modal";
import { PressableWithScaleEffect } from "./PressableWithScaleEffect";
import { Text } from "./Text";

import { useCustomThemeContext } from "src/context/CustomThemeProvider";
import { ActivityType } from "src/services/database/models/ActivityModel";
import { activityTypeAttributes } from "src/utils/global";

const activityTypes = Object.entries(activityTypeAttributes);

type ActivityTypesModalProps = {
  onPress: (_type: ActivityType) => void;
};

export type ActivityTypesModalRef = {
  close: () => void;
  open: () => void;
};

export const ActivityTypesModal = forwardRef<
  ActivityTypesModalRef,
  ActivityTypesModalProps
>(({ onPress }, ref) => {
  const refModal = useRef<ModalRef>(null);

  const { theme } = useCustomThemeContext();

  useImperativeHandle(ref, () => ({
    close: () => {
      refModal.current?.close();
    },
    open: () => {
      refModal.current?.open();
    }
  }));

  return (
    <Modal containerClassName="px-4 space-y-6" ref={refModal}>
      <View className="flex-row justify-between px-2">
        <Text className="font-bold text-lg">New Activity</Text>
        <TouchableOpacity onPress={refModal.current?.close}>
          <Ionicons name="close-circle" size={24} color={theme.colors.border} />
        </TouchableOpacity>
      </View>
      <View className="flex-row flex-wrap">
        {activityTypes.map(([type, attributes]) => (
          <PressableWithScaleEffect
            className="w-1/3 p-2"
            key={type}
            onPress={() =>
              !attributes.commingSoon && onPress(type as ActivityType)
            }
          >
            <Card.Container className="flex-col" color={attributes.color}>
              <View className="items-center">
                <View>
                  <Card.RoundedSquare withBorder>
                    <Card.Title>{attributes.emoji}</Card.Title>
                  </Card.RoundedSquare>
                  {attributes.commingSoon && (
                    <View
                      className="absolute px-1 py-0.5 -right-1 rounded-md -top-1"
                      style={{
                        backgroundColor: colors[attributes.color][500]
                      }}
                    >
                      <Card.Caption
                        className=" text-[10px]"
                        style={{ color: colors[attributes.color][100] }}
                      >
                        soon
                      </Card.Caption>
                    </View>
                  )}
                </View>
                <Card.Title className="font-medium text-sm" numberOfLines={1}>
                  {attributes.title}
                </Card.Title>
              </View>
            </Card.Container>
          </PressableWithScaleEffect>
        ))}
      </View>
    </Modal>
  );
});

ActivityTypesModal.displayName = "ActivityTypesModal";
