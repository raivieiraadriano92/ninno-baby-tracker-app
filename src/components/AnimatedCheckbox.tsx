import { ComponentProps, FunctionComponent, useEffect } from "react";

import Ionicons from "@expo/vector-icons/Ionicons";
import Animated, {
  FadeIn,
  FadeOut,
  interpolateColor,
  LinearTransition,
  useAnimatedStyle,
  useSharedValue,
  withTiming
} from "react-native-reanimated";
import colors from "tailwindcss/colors";

import { PressableWithScaleEffect } from "./PressableWithScaleEffect";
import { Text } from "./Text";

type AnimatedCheckboxProps = ComponentProps<typeof PressableWithScaleEffect> & {
  color: keyof typeof colors;
  isSelected: boolean;
  textClassName?: string;
  title: string;
};

export const AnimatedCheckbox: FunctionComponent<AnimatedCheckboxProps> = ({
  className,
  color,
  isSelected,
  style,
  textClassName,
  title,
  ...props
}) => {
  const progress = useSharedValue(0);

  const animatedStyle = useAnimatedStyle(() => ({
    backgroundColor: interpolateColor(
      progress.value,
      [0, 1],
      [colors.white, colors[color][100]]
    ),
    borderColor: interpolateColor(
      progress.value,
      [0, 1],
      [colors.neutral[100], colors[color][500]]
    )
  }));

  useEffect(() => {
    if (isSelected) {
      progress.value = withTiming(1);
    } else {
      progress.value = withTiming(0);
    }
  }, [isSelected, progress]);

  return (
    <PressableWithScaleEffect
      className={`border-[1px] flex-row items-center py-1 px-3 rounded-full space-x-1 ${className}`}
      layout={LinearTransition}
      style={[animatedStyle, style]}
      {...props}
    >
      {isSelected && (
        <Animated.View
          entering={FadeIn.delay(100)}
          exiting={FadeOut}
          layout={LinearTransition}
        >
          <Ionicons
            name="checkmark-circle"
            size={16}
            color={colors[color][500]}
          />
        </Animated.View>
      )}
      <Animated.View layout={LinearTransition}>
        <Text
          className={`capitalize font-medium text-sm ${textClassName}`}
          style={{
            color: isSelected ? colors[color][500] : colors.neutral[300]
          }}
        >
          {title}
        </Text>
      </Animated.View>
    </PressableWithScaleEffect>
  );
};
