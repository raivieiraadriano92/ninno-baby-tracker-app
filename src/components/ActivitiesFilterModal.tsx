import { forwardRef, useImperativeHandle, useState } from "react";

import { BlurView } from "expo-blur";
import { Modal, View } from "react-native";
import { SafeAreaProvider, SafeAreaView } from "react-native-safe-area-context";

import { AnimatedCheckbox } from "./AnimatedCheckbox";
import { Button } from "./Button";
import { Text } from "./Text";

import { useCustomThemeContext } from "src/context/CustomThemeProvider";

type ActivitiesFilterModalProps = {
  //
};

export type ActivitiesFilterModalRef = {
  open: () => void;
};

export const ActivitiesFilterModal = forwardRef<
  ActivitiesFilterModalRef,
  ActivitiesFilterModalProps
>(({}, ref) => {
  const { theme } = useCustomThemeContext();

  const [state, setState] = useState({
    period: "Weekly",
    isVisible: false
  });

  useImperativeHandle(ref, () => ({
    open: () => {
      setState((prev) => ({
        ...prev,
        isVisible: true
      }));
    }
  }));

  const close = () => {
    setState((prev) => ({
      ...prev,
      isVisible: false
    }));
  };

  const apply = () => {
    setState((prev) => ({
      ...prev,
      isVisible: false
    }));
  };

  return (
    <Modal animationType="fade" transparent visible={state.isVisible}>
      <BlurView
        className="flex-1"
        experimentalBlurMethod="dimezisBlurView"
        intensity={100}
      >
        <SafeAreaProvider>
          <SafeAreaView className="flex-1 justify-end p-6 space-y-6">
            <Text className="font-bold text-2xl">Filters</Text>
            <View className="flex-row space-x-2">
              {["Weekly", "Monthly", "7-day", "30-day"].map((period) => (
                <AnimatedCheckbox
                  color={theme.colorTokens.primary}
                  isSelected={state.period === period}
                  key={period}
                  onPress={() =>
                    setState((prev) => ({
                      ...prev,
                      period: period
                    }))
                  }
                  title={period}
                />
              ))}
            </View>
            <View className="space-y-4">
              <Button onPress={apply} title="Apply" />
              <Button onPress={close} title="Cancel" variant="link" />
            </View>
          </SafeAreaView>
        </SafeAreaProvider>
      </BlurView>
    </Modal>
  );
});

ActivitiesFilterModal.displayName = "ActivitiesFilterModal";
