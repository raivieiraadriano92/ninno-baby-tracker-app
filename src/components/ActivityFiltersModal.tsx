import { forwardRef, useImperativeHandle, useRef } from "react";

import { View } from "react-native";
import colors from "tailwindcss/colors";

import { AnimatedCheckbox } from "./AnimatedCheckbox";
import { Button } from "./Button";
import { Modal, ModalRef } from "./Modal";
import { Text } from "./Text";

import { useCustomThemeContext } from "src/context/CustomThemeProvider";
import { ActivityTypeGroup } from "src/services/database/models/ActivityModel";
import {
  ActivityFiltersPeriod,
  useActivityFiltersStore
} from "src/store/activityFiltersStore";
import { activityTypeGroupAttributes } from "src/utils/global";

type ActivityFiltersModalProps = {
  //
};

export type ActivityFiltersModalRef = {
  open: () => void;
};

export const ActivityFiltersModal = forwardRef<
  ActivityFiltersModalRef,
  ActivityFiltersModalProps
>(({}, ref) => {
  const activityFiltersStore = useActivityFiltersStore();

  const refModal = useRef<ModalRef>(null);

  const { theme } = useCustomThemeContext();

  const apply = () => {
    refModal.current?.close();
  };

  useImperativeHandle(ref, () => ({
    open: () => refModal.current?.open()
  }));

  return (
    <Modal ref={refModal}>
      <View className="space-y-6">
        <Text className="font-bold text-lg">Filters</Text>
        <View className="border-neutral-100 border-[1px] p-4 rounded-2xl space-y-4">
          <View className="space-y-2">
            <Text
              className="text-sm"
              style={{ color: colors[theme.colorTokens.primary][500] }}
            >
              Period
            </Text>
            <View className="flex-row flex-wrap" style={{ gap: 8 }}>
              {Object.values(ActivityFiltersPeriod).map((period) => (
                <AnimatedCheckbox
                  color={theme.colorTokens.primary}
                  isSelected={activityFiltersStore.period === period}
                  key={period}
                  onPress={() => activityFiltersStore.setPeriod(period)}
                  title={period}
                />
              ))}
            </View>
          </View>
          <View className="bg-neutral-100 h-[1px]" />
          <View className="space-y-2">
            <Text
              className="text-sm"
              style={{ color: colors[theme.colorTokens.primary][500] }}
            >
              Activity
            </Text>
            <View className="flex-row flex-wrap" style={{ gap: 8 }}>
              <AnimatedCheckbox
                color={theme.colorTokens.primary}
                isSelected={activityFiltersStore.activityTypeGroup === "All"}
                onPress={() => activityFiltersStore.setActivityTypeGroup("All")}
                title="All"
              />
              {Object.entries(activityTypeGroupAttributes).map(
                ([group, { color, emoji, title }]) => (
                  <AnimatedCheckbox
                    color={color}
                    isSelected={
                      activityFiltersStore.activityTypeGroup === group
                    }
                    key={group}
                    onPress={() =>
                      activityFiltersStore.setActivityTypeGroup(
                        group as ActivityTypeGroup
                      )
                    }
                    title={`${emoji} ${title}`}
                  />
                )
              )}
            </View>
          </View>
        </View>
        <View className="space-y-4">
          <Button
            customColors={[
              colors[theme.colorTokens.primary][500],
              colors[theme.colorTokens.primary][600]
            ]}
            onPress={apply}
            title="Apply"
          />
        </View>
      </View>
    </Modal>
  );
});

ActivityFiltersModal.displayName = "ActivityFiltersModal";
