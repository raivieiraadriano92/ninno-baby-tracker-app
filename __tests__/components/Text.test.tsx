import { render } from "@testing-library/react-native";

import { Text } from "src/components/Text";

describe("<Text />", () => {
  test("Snapshot", () => {
    const tree = render(<Text>My Text</Text>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
