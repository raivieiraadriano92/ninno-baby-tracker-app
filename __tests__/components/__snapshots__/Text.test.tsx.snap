// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Snapshot 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "400",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      {
        "color": "#000",
      },
    ]
  }
>
  My Text
</Text>
`;
